### PR TITLE
Add a proxy test

### LINKED ISSUE #9533: Write an e2e test for the node proxy
Context: #9029

### PR DESCRIPTION
**Pull request by lavalamp**
Fixes #9533

### CODE REVIEW

**Comment by lavalamp** [comment]
Note this depends on a fix in #9592.

File: test/e2e/framework.go [hunk 1]
```diff
@@ -103,3 +103,47 @@ func (f *Framework) WaitForPodRunning(podName string) error {
 func (f *Framework) TestContainerOutput(scenarioName string, pod *api.Pod, expectedOutput []string) {
 	testContainerOutputInNamespace(scenarioName, f.Client, pod, expectedOutput, f.Namespace.Name)
 }
+
+func (f *Framework) WaitForEndpoints(serviceName string) error {
```
Comment (brendandburns) [inline test/e2e/framework.go]: Document what this does.  The name seems to indicate it waits for all to be ok, but the implementation just waits for one to be ok.

File: test/e2e/proxy.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+})
+
+func proxyContext(version string) {
+	f := NewFramework("proxy")
+	prefix := "/api/" + version
+
+	It("should proxy logs on node with explicit kubelet port", func() {
+		node, err := pickNode(f.Client)
+		Expect(err).NotTo(HaveOccurred())
+		body, err := f.Client.Get().AbsPath(prefix + "/proxy/nodes/" + node + ":10250/logs/").Do().Raw()
```
Comment (brendandburns) [inline test/e2e/proxy.go]: nit, a number of ```Path(...).``` on different lines will make this easier to read.

File: test/e2e/proxy.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+						Name:       "portname2",
+						Port:       81,
+						TargetPort: util.NewIntOrStringFromInt(162),
+					},
+				},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			f.Client.Services(service.Namespace).Delete(service.Name)
```
Comment (brendandburns) [inline test/e2e/proxy.go]: nit: why not just defer this call?

**Comment by brendandburns** [comment]
Small comments.  Shippable is a flake, hit it again.

File: test/e2e/framework.go (hunk 1 above)
Comment (lavalamp) [inline test/e2e/framework.go]: Will fix. It's confusing because we named a single object "Endpoints"...

File: test/e2e/proxy.go (hunk 3 above)
Comment (lavalamp) [inline test/e2e/proxy.go]: Because I should be checking the error... :)

File: test/e2e/proxy.go (hunk 2 above)
Comment (lavalamp) [inline test/e2e/proxy.go]: Yeah, but then the trailing / gets eaten and it doesn't work. Will add a comment.

**Comment by lavalamp** [comment]
Updated. This won't pass e2e until #9592 is merged.

**Comment by quinton-hoole** [comment]
LGTM.  #9592 has been merged.  Merging this as it's a very useful signal and the change is limited to a single test, and easy to roll back if required.  I'll keep an eye on it over the weekend.

**Comment by quinton-hoole** [comment]
@k8s-bot test this please

**Pull request merged**